### DISCUSSION TITLE
👽️ `CPAC.utils.interfaces.function.Function` vs `nipype.interfaces.Function`

### DISCUSSION THREAD

**Discussion by shnizzedy** [FCP-INDI/C-PAC#1848] [Ideas]
> In [C-PAC's variant](https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/utils/interfaces/function/function.py#L14-L170) of [Nipype's Function Node Interface](https://github.com/nipy/nipype/blob/071523a5069b52689ed27f995f1e2bea5b6e0402/nipype/interfaces/utility/wrappers.py#L24-L159), we have the option to be a little more pythonic by passing https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/utils/interfaces/function/function.py#L58-L60

_Originally posted by @shnizzedy in https://github.com/FCP-INDI/C-PAC/pull/1846#pullrequestreview-1184984388_

Every time I think something is unique in our version and check theirs, it's there. Do we still have a need to maintain our own flavor of `Function` interface?

**Comment by nx10** [comment]
It might also be worth discussing whether a list if dynamic imports is even worth the additional overhead and complexity (compared to explicit function-local imports).

**Comment by shnizzedy** [comment]
Since opening this discussion, I've made changes in our `Function` interface, so this is no longer relevant.
